packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -0,0 +1,91 @@
+import fetch, { RequestInit } from 'node-fetch';
+
+type RequestMethods = 'post' | 'get';
+type RequestParams = {
+	method: 'post' | 'get';
+	headers: GenericDict;
+	body?: GenericDict;
+};
+type GenericDict = { [ key: string ]: string | number };
+
+interface Site {
+	ID: string;
+	name: string;
+	description: string;
+	URL: string;
+	site_owner: string;
+}
+interface MySitesResponse {
+	sites: Site[];
+}
+
+const BASE_URL = 'https://public-api.wordpress.com/rest/v1.1/';
+
+/**
+ * Client to interact with the WordPress.com REST API.
+ */
+export class RestAPIClient {
+	readonly bearerToken: string;
+
+	/**
+	 * Constructs an instance of the API client.
+	 */
+	constructor( bearerToken: string ) {
+		this.bearerToken = bearerToken;
+	}
+
+	/**
+	 * Builds and returns the request parameters.
+	 *
+	 * @param {RequestMethods} method Supported request methods.
+	 * @param {GenericDict} [body] Request body.
+	 * @returns {RequestParams} Completed request parameter.
+	 */
+	private buildRequestParams( method: RequestMethods, body?: GenericDict ): RequestParams {
+		const params: RequestParams = {
+			method: method,
+			headers: {
+				Authorization: `Bearer ${ this.bearerToken }`,
+				'Content-Type': 'application/json',
+			},
+		};
+
+		// POST requests require a body.
+		if ( method === 'post' ) {
+			if ( ! body ) {
+				throw new Error( 'Cannot build a POST request with undefined body.' );
+			}
+			params[ 'body' ] = body;
```

Comment:
Quick FYI -- I think the fetch API expects JSON bodies to be strings created from `JSON.stringify`

Reply:
Hmm, it seems to have been working fine without the `JSON.stringify` method call, but I'll make the change nonetheless. Thanks.

Reply:
Hmm, maybe this a change they made to the node version? Or in a new version they no longer require that? I just remember this from prior experience and the docs.

Reply:
Perhaps - or it might be because I don't have POST methods with JSON implemented, that may be why!